src/utils/index.ts

```diff
@@ -54,7 +54,16 @@
   args?: Inputs,
   optionalParams?: Inputs,
 ) {
-  return code.replace(/{{(.*?)}}/g, (match) => {
+  // Conditionals.
+  code = code.replace(/{{#(.*?)}}(.*){{\/\1}}/g, (match, name, innerCode) => {
+    if (args?.[name] || optionalParams?.[name]) {
+      return innerCode;
+    }
+    return '';
+  });
```

Comment:
@housseindjirdeh @flashdesignory Not sure whether we need to do something about this? Do you think it's relevant to address?

Reply:
hmm, sounds like it might mostly be an issue with larger input, but maybe still something we should fix. 
what was the reasoning to change the regular expression? Was it to avoid filtering out nullisch values afterwards, or did the consent object require us to change this?

Reply:
It's not a change, this is an entirely new regular expression, to account for if constructs using Mustache syntax (see PR description for more context).

I'm not sure I understand the CodeQL complaint here or how we could fix it. 🤔 FWIW it all works as expected as shown by the new tests.

Reply:
the regex makes sense to me and seems correct (although I'm not a regex master,lol). 

Maybe we could limit the string input length? or we could throw an error earlier if it starts with '{{{{#' for some reason?